presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -857,7 +833,7 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, Context context)
 
                 source = node.getSource().accept(this, context.withPreferredProperties(PreferredProperties.partitioned(ImmutableSet.copyOf(sourceSymbols))));
 
-                if (source.getProperties().isNodePartitionedOn(sourceSymbols) && !(source.getProperties().isSingleNode() && distributedJoins)) {
```

Comment:
It was entirely redundant here because it's already nested in an if statement that checks that distributedJoins is ture